x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/utils.ts

```diff
@@ -35,12 +34,12 @@ export const convertRulesFilterToKQL = ({
 }: FilterOptions): string => {
   const filters: string[] = [];
 
-  if (showCustomRules) {
-    filters.push(`alert.attributes.tags: "${INTERNAL_IMMUTABLE_KEY}:false"`);
-  }
-
-  if (showElasticRules) {
-    filters.push(`alert.attributes.tags: "${INTERNAL_IMMUTABLE_KEY}:true"`);
+  if (showCustomRules && showElasticRules) {
+    filters.push('alert.attributes.params.immutable: (true OR false)');
```

Comment:
`alert.attributes.params.immutable: (true OR false)` query looks redundant as it matches all rules. Can we just omit it?

Reply:
done